src/js/profile/mobile/widget/Calendar.js

```diff
@@ -0,0 +1,295 @@
+ */
+(function (document, ns) {
+	"use strict";
+	//>>excludeStart("tauBuildExclude", pragmas.tauBuildExclude);
+	define(
+		[
+			"../../../core/engine",
+			"../../../core/util/DOM/attributes",
+			"../../../core/event",
+			"../widget"
+		],
+		function () {
+			//>>excludeEnd("tauBuildExclude");
+			var utilsObject = ns.util.object,
+				engine = ns.engine,
+				events = ns.event,
+				days = ["MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"],
+				fullNameMonth = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"],
+				today = new Date(),
+				todayYear = today.getFullYear(),
+				todayMonth = today.getMonth() + 1,
+
+				Calendar = function () {
+					this.options = utilsObject.merge({}, defaultOptions);
+				},
+
+				defaultOptions = {
+					pastSelection: false
+				},
+
+				BaseWidget = ns.widget.BaseWidget,
+				prototype = new BaseWidget();
+
+			Calendar.prototype = prototype;
+
+			/**
+			* Init widget
+			* @method _init
+			* @param {HTMLElement} element
+			* @return {HTMLElement} Returns built element
+			* @member ns.widget.mobile.Calendar
+			* @protected
+			*/
+			prototype._init = function (element) {
+				var self = this;
+
+				if (!element.getAttribute("value")) {
+					element.setAttribute("value", self.options.value);
+				}
+
+				self._buildCalendar();
+
+				return element;
+			};
+
+			/**
+			* Draw a calendar on the table
+			* @method _buildCalendar
+			* @member ns.widget.mobile.Calendar
+			* @protected
+			*/
+			prototype._buildCalendar = function () {
+				var firstDate = new Date(todayYear, todayMonth - 1, 0),
+					lastDate = new Date(todayYear, todayMonth, 0),
+					day = firstDate.getDay(),
+					prevLastDate = lastDate.getDate(),
+					week = Math.ceil(prevLastDate / 7) + 1,
+					leftDays = 7,
+					setDays = 1,
+					nextMonthDate = 1,
+					cell = null,
+					calendar = document.getElementById("calendar"),
+					marginRow = calendar.insertRow(),
+					idx,
+					row;
+
+				marginRow.style.height = "7px";
+
+				for (idx = 1; idx < week + 1; idx++) {
+					row = calendar.insertRow();
+
+					while (day != 0) { // prev disabled days.
+						day = day - 1;
+						leftDays = leftDays - 1;
+						cell = row.insertCell();
+						cell.innerHTML = prevLastDate - day;
+						cell.style.opacity = "10%";
+					}
+					while (leftDays != 0) {
+						if (setDays > prevLastDate) { // next disabled days.
+							cell = row.insertCell();
+							cell.innerHTML = nextMonthDate;
+							cell.style.opacity = "40%";
+							leftDays = leftDays - 1;
+							nextMonthDate = nextMonthDate + 1;
+						} else { // current enabled days.
+							cell = row.insertCell();
+							cell.innerHTML = setDays;
+							cell.style.opacity = "100%";
+							setDays = setDays + 1;
+							leftDays = leftDays - 1;
+						}
+					}
+					leftDays = 7;
+				}
+
+				document.getElementById("yearMonth").innerHTML = fullNameMonth[todayMonth - 1] + " " + todayYear;
```

Comment:
Using "ID" for identify element is good for app developer but framework developer cannot use ID.
There may be a conflict, better use `class` and `querySelector` to select element.

Reply:
I change getElementById to querySelector.

done.